rdflib/extras/infixowl.py

```diff
@@ -195,6 +196,12 @@ def __rlshift__(self, other):
     def __rshift__(self, other):
         return self.function(other)
 
+    def __rmatmul__(self, other):
+        return Infix(lambda x, self=self, other=other: self.function(other, x))
+
+    def __matmul__(self, other):
+        return self.function(other)
+
```

Comment:
Are these really needed? As far as I understand this makes `A @ B @ C` do the same as `A | B | C` and `A << B >> C`.

I think this Infix OWL has diverged from the [spec](https://www.w3.org/TR/owl-semantics/syntax.html) somewhat but [@ is reserved in there for language tags](https://www.w3.org/TR/owl-semantics/syntax.html#2.2) I think so it may be best to leave it for that.

I'm not an expert at OWL manchester syntax, so I may be missing something here - maybe @ashleysommer or @nicholascar has some input about it.

Reply:
As I read it, when gromgull migrated the SPARQL paths processor `paths.py` out of the sparql module back in 2103, the change from monkeypatching Path to directly defining Path methods [definitions](https://github.com/RDFLib/rdflib/blob/43d86224382d64fba2e5e17acedea1704631e097/rdflib/paths.py#L199) overrode the `|` (`__or__` and `__ror__`)  infixOwl delimiters causing exceptions in the infixOwl doctests. Hence my editing the docstrings to use the existing alternative implementation of `<<` and `>>` (`__ rshift` and `__rlshift__`) instead.

I subsequently discovered that unfortunately, SPARQL* uses the Turtle* double-anglebrackets and, anticipating that addition to `Path`, I tried to find an alternative to use with infixOwl. As far as I could ascertain, matmul was the only remaining option but yes, I guess it will  clash with `@` as lang tag so there doesn't appear to be any ready solution for enabling infixOwl to work in RDFLib with the “colloquial programming style” described in the infixOwl docstrings. 

>  think this Infix OWL has diverged from the spec somewhat

TBH, those specific infix operators aren't part of the spec, they were added by Chimezie as part of FuXi and weren't in the RDFLib 2.4 of the time. I'm not going to spend time arguing for the retention of infixOwl if the implementation has aged to a degree that it's now basically out of scope, other than to observe the irony of Harold (funowl) Solbrig's [respect for it](https://github.com/hsolbrig/funowl#other-packages).

Reply:
I'm fine kind of with `@` being used for this, like you said there is already some deviations and using `@` seems like a minor one, though still just want a double check from @nicholascar or someone else on this point.

Reply:
I think it sensible to avoid `<<` & `>>` as they are indeed very likely to be implemented for Turtle-Start soon, as noted above.

We can always expect to come up with notation/symbol clashes from time-to-time given that we are dealing with:

* Python
* SPARQL
* multiple RDF syntaxes

..all in one.

So, I think it fine to implement `@` here for now and just have to deal with any follow-on issues later, if further syntax conflicts appear.

I am keen to get (@hsolbrig)'s Manchester Syntax into RDFLib (perhaps in 7.0.0??) but patching up preexisting packages first seems sensible.

Reply:
I replied inline about this elsewhere: TL;DR: fine for now.